### PR TITLE
OCPCLOUD-2278: Add kube-rbac-proxy container & ensure metrics are only available via HTTPS

### PR DESCRIPTION
**Pull request by racheljpg**
This is a PR to get feedback and run tests on my changes for [OCPCLOUD-2278](https://issues.redhat.com//browse/OCPCLOUD-2278) - will add a more in depth description once we're sure it works as expected.

### CODE REVIEW

File: manifests/0000_26_cloud-controller-manager-operator_11_deployment.yaml [hunk 1]
```diff
@@ -138,4 +166,4 @@ spec:
       - name: host-etc-kube
         hostPath:
           path: /etc/kubernetes
-          type: Directory
+          type: Directory
```
Comment (JoelSpeed) [inline manifests/0000_26_cloud-controller-manager-operator_11_deployment.yaml:169]: Nit, undo this removal of new line char

File: manifests/0000_26_cloud-controller-manager-operator_11_deployment.yaml [hunk 2]
```diff
@@ -46,10 +46,10 @@ spec:
           --leader-elect-retry-period=26s \
           --leader-elect-resource-namespace=openshift-cloud-controller-manager-operator \
           "--images-json=/etc/cloud-controller-manager-config/images.json" \
-          --metrics-bind-address=:9258 \
+          --metrics-bind-address=:9257 \
```
Comment (JoelSpeed) [inline manifests/0000_26_cloud-controller-manager-operator_11_deployment.yaml:49]: This needs to be bound on the loopback only
```suggestion
          --metrics-bind-address=127.0.0.1:9257 \
```

File: manifests/0000_26_cloud-controller-manager-operator_11_deployment.yaml [hunk 3]
```diff
@@ -103,6 +103,34 @@ spec:
...
+        - --tls-private-key-file=/etc/tls/private/tls.key
+        - --config-file=/etc/kube-rbac-proxy/config-file.yaml
+        - --tls-cipher-suites=TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305
+        - --logtostderr=true
+        - --v=3
+        image: quay.io/openshift/origin-kube-rbac-proxy:4.2.0
+        imagePullPolicy: IfNotPresent
+        name: kube-rbac-proxy
+        ports:
+        - containerPort: 9192
```
Comment (JoelSpeed) [inline manifests/0000_26_cloud-controller-manager-operator_11_deployment.yaml:119]:
```suggestion
        - containerPort: 9258
```

File: manifests/0000_26_cloud-controller-manager-operator_11_deployment.yaml [hunk 4]
```diff
@@ -103,6 +103,34 @@ spec:
           - mountPath: /etc/kubernetes
             name: host-etc-kube
             readOnly: true
+      - args:
+        - --secure-listen-address=0.0.0.0:9192
+        - --upstream=http://127.0.0.1:9191/
```
Comment (JoelSpeed) [inline manifests/0000_26_cloud-controller-manager-operator_11_deployment.yaml:108]:
```suggestion
        - --secure-listen-address=0.0.0.0:9258
        - --upstream=http://127.0.0.1:9257/
```

File: manifests/0000_26_cloud-controller-manager-operator_11_deployment.yaml (hunk 1 above)
Comment (racheljpg) [reply manifests/0000_26_cloud-controller-manager-operator_11_deployment.yaml]: Done!

File: manifests/0000_26_cloud-controller-manager-operator_11_deployment.yaml (hunk 4 above)
Comment (racheljpg) [reply manifests/0000_26_cloud-controller-manager-operator_11_deployment.yaml]: Thanks very much for the review Joel, I've implemented your changes and a volume I was missing.

File: manifests/0000_26_cloud-controller-manager-operator_19_metrics-service.yaml [hunk 5]
```diff
@@ -0,0 +1,22 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: cloud-manager-operator
+  namespace: openshift-cloud-manager-operator
```
Comment (JoelSpeed) [inline manifests/0000_26_cloud-controller-manager-operator_19_metrics-service.yaml:5]:
```suggestion
  namespace: openshift-cloud-controller-manager-operator
```

File: manifests/0000_26_cloud-controller-manager-operator_19_metrics-service.yaml [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  annotations:
+    include.release.openshift.io/ibm-cloud-managed: "true"
+    include.release.openshift.io/self-managed-high-availability: "true"
+    include.release.openshift.io/single-node-developer: "true"
+    service.alpha.openshift.io/serving-cert-secret-name: cloud-manager-operator-tls
```
Comment (JoelSpeed) [inline manifests/0000_26_cloud-controller-manager-operator_19_metrics-service.yaml:10]:
```suggestion
    service.alpha.openshift.io/serving-cert-secret-name: cloud-controller-manager-operator-tls
```

File: manifests/0000_26_cloud-controller-manager-operator_11_deployment.yaml [hunk 7]
```diff
@@ -139,3 +167,9 @@ spec:
         hostPath:
           path: /etc/kubernetes
           type: Directory
+      - configMap:
+          name: kube-rbac-proxy
+        name: auth-proxy-config
+      - name: cloud-manager-operator-tls
+        secret:
+          secretName: cloud-manager-operator-tls
```
Comment (JoelSpeed) [inline manifests/0000_26_cloud-controller-manager-operator_11_deployment.yaml:175]:
```suggestion
          secretName: cloud-controller-manager-operator-tls
```

File: manifests/0000_26_cloud-controller-manager-operator_11_deployment.yaml [hunk 8]
```diff
@@ -139,3 +167,9 @@ spec:
         hostPath:
           path: /etc/kubernetes
           type: Directory
+      - configMap:
+          name: kube-rbac-proxy
+        name: auth-proxy-config
+      - name: cloud-manager-operator-tls
```
Comment (JoelSpeed) [inline manifests/0000_26_cloud-controller-manager-operator_11_deployment.yaml:173]:
```suggestion
      - name: cloud-controller-manager-operator-tls
```

File: manifests/0000_26_cloud-controller-manager-operator_11_deployment.yaml [hunk 9]
```diff
@@ -103,6 +103,34 @@ spec:
...
+        terminationMessagePolicy: File
+        resources:
+          requests:
+            memory: 20Mi
+            cpu: 10m
+        volumeMounts:
+        - mountPath: /etc/kube-rbac-proxy
+          name: auth-proxy-config
+        - mountPath: /etc/tls/private
+          name: cloud-manager-operator-tls
```
Comment (JoelSpeed) [inline manifests/0000_26_cloud-controller-manager-operator_11_deployment.yaml:133]:
```suggestion
          name: cloud-controller-manager-operator-tls
```

**Review by JoelSpeed — COMMENTED** [review]
/lgtm

**Pull request merged**